t/io/eintr.t

```diff
@@ -73,18 +73,27 @@ sub fresh_io {
     undef $in; undef $out; # use fresh handles each time
     pipe $in, $out;
     $sigst = "";
+    $pipe_buf_err = "";
 
     # This used to be 1_000_000, but on Linux/ppc64 (POWER7) this kept
     # consistently failing. At exactly 0x100000 it started passing
     # again. Now we're asking the kernel what the pipe buffer is, and if
     # that fails, hoping this number is bigger than any pipe buffer.
     $pipe_buf_size = eval {
         use Fcntl qw(F_GETPIPE_SZ);
-        fcntl($out, F_GETPIPE_SZ, 0);
+        # When F_GETPIPE_SZ isn't implemented then fcntl() raises an exception:
+        #   "Your vendor has not defined Fcntl macro F_GETPIPE_SZ ..."
+        # When F_GETPIPE_SZ is implemented then errors are still possible
+        # (EINVAL, EBADF, ...). These are not exceptions (i.e. these don't die)
+        # but instead these set $! and make fcntl() return undef.
+        fcntl($out, F_GETPIPE_SZ, 0) or die "$!\n";
     };
     if ($@ or not $pipe_buf_size) {
+        my $err = $@;;
+        chomp $err;
         $pipe_buf_size = 0xfffff;
-        diag("fcntl F_GETPIPE_SZ failed, failing back to $pipe_buf_size");
+        $pipe_buf_err = "fcntl F_GETPIPE_SZ failed" . ($err ? " ($err)" : "") .
+                        ", failing back to $pipe_buf_size";
```

Comment:
I don't understand "failing back"

Reply:
its a fancy way of saying "defaulting to $pipe_buf_size". :-)